File: backend/HeartmadeCandles/HeartmadeCandles.Order.Core/Models/Order.cs
﻿namespace HeartmadeCandles.Order.Core.Models;

public class Order
{
    public string? Id { get; init; }

    public required string BasketId { get; init; }

    public Basket? Basket { get; init; }

    public Feedback? Feedback { get; init; }

    public OrderStatus Status { get; init; }

    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; init; } = DateTime.UtcNow;
}

File: frontend/heartmage-candles/src/typesV2/shared/Order.ts
import { Basket } from '../order/Basket';
import { Feedback } from '../order/Feedback';
import { OrderStatus } from '../order/OrderStatus';

export interface Order {
  id?: string;
  basketId: string;
  basket?: Basket;
  feedback: Feedback;
  status?: OrderStatus;
  createdAt: Date;
  updatedAt: Date;
}
